### PR TITLE
Standardize naming in generic storage watcher tests

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/109831

```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig api-machinery

/assign @stevekuznetsov 
/cc @tallclair

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go [hunk 1]
```diff
@@ -44,65 +44,76 @@ func RunTestWatch(ctx context.Context, t *testing.T, store storage.Interface) {
 // - update should trigger Modified event
 // - update that gets filtered should trigger Deleted event
 func testWatch(ctx context.Context, t *testing.T, store storage.Interface, recursive bool) {
-	podFoo := &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}
-	podBar := &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "bar"}}
+	basePod := &example.Pod{
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go:47]: These changes are effectively combinining https://github.com/kubernetes/kubernetes/pull/113696 with https://github.com/kubernetes/kubernetes/pull/113666 for watcher tests.

**Review by stevekuznetsov — APPROVED** [review]

File: staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go [hunk 2]
```diff
@@ -44,65 +44,76 @@ func RunTestWatch(ctx context.Context, t *testing.T, store storage.Interface) {
...
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			w, err := store.Watch(ctx, tt.key, storage.ListOptions{ResourceVersion: "0", Predicate: tt.pred, Recursive: recursive})
+			watchKey := fmt.Sprintf("pods/%s", tt.namespace)
+			key := watchKey + "/foo"
+			if !recursive {
+				watchKey = key
+			}
+
+			w, err := store.Watch(ctx, watchKey, storage.ListOptions{ResourceVersion: "0", Predicate: tt.pred, Recursive: recursive})
```
Comment (logicalhan) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go:105]: nice this is way more readable

File: staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go [hunk 3]
```diff
@@ -44,65 +44,76 @@ func RunTestWatch(ctx context.Context, t *testing.T, store storage.Interface) {
...
 			if err != nil {
 				t.Fatalf("Watch failed: %v", err)
 			}
 			var prevObj *example.Pod
 			for _, watchTest := range tt.watchTests {
 				out := &example.Pod{}
-				key := tt.key
-				if recursive {
-					key = key + "/item"
-				}
```
Comment (logicalhan) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go:102]: I always hated this bit.

**Review by logicalhan — COMMENTED** [review]
/lgtm

File: staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go (hunk 3 above)
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go:102]: :)

**Comment by wojtek-t** [comment]
/milestone v1.26

[This was tagged before code-freeze, and in addition this is test-only change.]

**Pull request merged**